web/client/plugins/Details.jsx

```diff
@@ -6,48 +6,118 @@
+    component: connect((state) => ({
+        show: get(state, "controls.details.enabled"),
+        loading: loadingSelector(state),
+        loadFlags: loadFlagsSelector(state),
+        mapInfo: mapInfoSelector(state),
+        canEdit: isLoggedIn(state) && get(mapInfoSelector(state), 'canEdit'),
+        settings: settingsSelector(state),
         map: mapFromIdSelector(state, mapIdSelector(state)),
-        dockStyle: mapLayoutValuesSelector(state, {height: true}),
-        detailsText: currentMapDetailsTextSelector(state)
+        content: contentSelector(state),
+        editedContent: editedContentSelector(state),
+        contentChanged: contentChangedSelector(state),
+        editedSettings: editedSettingsSelector(state),
+        editing: editingSelector(state),
+        dockStyle: mapLayoutValuesSelector(state, {height: true})
     }), {
-        onClose: closeDetailsPanel
-    })(assign(require('../components/details/DetailsPanel'), {
+        onSave: save,
+        onEdit: edit.bind(null, 'content'),
+        onEditSettings: edit.bind(null, 'settings'),
+        onCancelEdit: cancelEdit,
+        onClose: setControlProperty.bind(null, 'details', 'enabled', false),
+        onSettingsChange: changeSetting,
+        onEditorUpdate: setEditedContent
+    }, (stateProps, dispatchProps, ownProps) => ({
+        ...stateProps,
+        ...ownProps,
+        ...dispatchProps,
+        editorProps: {
+            ...(stateProps.editorProps || {}),
+            ...(ownProps.editorProps || {}),
+            onUpdate: dispatchProps.onEditorUpdate
+        }
+    }))(Details),
+    containers: {
         BurgerMenu: {
             name: 'details',
             position: 1000,
             text: <Message msgId="details.title"/>,
             icon: <Glyphicon glyph="sheet"/>,
-            action: openDetailsPanel,
+            action: setControlProperty.bind(null, 'details', 'enabled', true),
+            selector: (state) => {
+                const mapId = mapIdSelector(state);
+                const detailsUri = mapId && mapInfoDetailsUriFromIdSelector(state, mapId);
+                const settings = settingsSelector(state) || {};
+                if (detailsUri && !settings.showAsModal) {
+                    return {};
+                }
+                return { style: {display: "none"} };
+            }
+        },
+        Toolbar: {
+            name: 'details',
+            position: 1,
+            priority: 1,
+            alwaysVisible: true,
+            doNotHide: true,
+            icon: <Glyphicon glyph="sheet"/>,
+            action: setControlProperty.bind(null, 'details', 'enabled', true),
             selector: (state) => {
                 const mapId = mapIdSelector(state);
                 const detailsUri = mapId && mapInfoDetailsUriFromIdSelector(state, mapId);
-                if (detailsUri) {
+                const settings = settingsSelector(state) || {};
+                if (detailsUri && settings.showAsModal) {
```

Comment:
I think showAsModal should not be necessary
```suggestion
                if (detailsUri) {
```